### PR TITLE
Delete PAL definitions of MS-specific __int... types

### PR DESCRIPTION
**Pull request by jkotas**
Also delete unused PAL methods with trasitive dependency on these types

### CODE REVIEW

**Comment by jkotas** [comment]
Re: https://github.com/dotnet/runtime/pull/102241#issuecomment-2112674696

File: src/coreclr/pal/inc/pal_mstypes.h [hunk 1]
```diff
@@ -271,35 +194,29 @@ typedef unsigned int *PUINT;
...
-typedef __int64 INT_PTR, *PINT_PTR;
-typedef unsigned __int64 UINT_PTR, *PUINT_PTR;
-typedef __int64 LONG_PTR, *PLONG_PTR;
-typedef unsigned __int64 ULONG_PTR, *PULONG_PTR;
-typedef unsigned __int64 DWORD_PTR, *PDWORD_PTR;
+typedef int64_t INT_PTR, *PINT_PTR;
+typedef uint64_t UINT_PTR, *PUINT_PTR;
+typedef int64_t LONG_PTR, *PLONG_PTR;
+typedef uint64_t ULONG_PTR, *PULONG_PTR;
+typedef uint64_t DWORD_PTR, *PDWORD_PTR;
```
Comment (huoyaoyuan) [inline src/coreclr/pal/inc/pal_mstypes.h:219]: Can these be changed to `intptr_t` and `uintptr_t` instead? This reduces ifdefs between 32bit and 64bit.

And also the `INTPTR_MAX` constant.

File: src/coreclr/pal/inc/pal_mstypes.h [hunk 2]
```diff
@@ -147,19 +125,6 @@ extern "C" {
 // Misc. type helpers
 ////////////////////////////////////////////////////////////////////////
 
-#ifdef _MSC_VER
-
-// MSVC's way of declaring large integer constants
-// If you define these in one step, without the _HELPER macros, you
-// get extra whitespace when composing these with other concatenating macros.
-#define I64_HELPER(x) x ## i64
-#define I64(x)        I64_HELPER(x)
```
Comment (huoyaoyuan) [inline src/coreclr/pal/inc/pal_mstypes.h:156]: There is `INT64_C` macro according to cppreference. Not sure of it's feasibility.

File: src/coreclr/pal/inc/pal_mstypes.h [hunk 3]
```diff
@@ -271,35 +194,29 @@ typedef unsigned int *PUINT;
...
+typedef int16_t INT16;
+typedef uint32_t UINT32, *PUINT32;
+typedef int32_t INT32, *PINT32;
+typedef uint64_t UINT64, *PUINT64;
+typedef int64_t INT64, *PINT64;
+
+typedef uint32_t ULONG32, *PULONG32;
+typedef int32_t LONG32, *PLONG32;
+typedef uint64_t ULONG64;
+typedef int64_t LONG64;
```
Comment (huoyaoyuan) [inline src/coreclr/pal/inc/pal_mstypes.h:209]: Are all these types actively used in CLR source code?

File: src/coreclr/dlls/mscordac/mscordac_unixexports.src [hunk 4]
```diff
@@ -60,8 +60,6 @@ nativeStringResourceTable_mscorrc
 #vsprintf_s
 #_snprintf_s
 #_vsnprintf_s
-#_itow_s
-#_i64tow_s
```
Comment (am11) [inline src/coreclr/dlls/mscordac/mscordac_unixexports.src:64]: One remaining instance in DAC:
```
src/coreclr/debug/daccess/daccess.cpp:    unsigned _int64 nStart, nEnd;
```

File: src/coreclr/pal/inc/pal_mstypes.h (hunk 2 above)
Comment (AaronRobinsonMSFT) [reply src/coreclr/pal/inc/pal_mstypes.h:156]: The `INT64_C` macro applies the "at least" logic and relying on the vagaries of a particular implementation with respect to "at least" is not something I think we should adopt.

File: src/coreclr/pal/inc/pal_mstypes.h (hunk 3 above)
Comment (AaronRobinsonMSFT) [reply src/coreclr/pal/inc/pal_mstypes.h:209]: Yes. This is a slow transition and changing these first and then pruning is the most appropriate option.

Comment (jkotas) [reply src/coreclr/pal/inc/pal_mstypes.h:209]: > Are all these types actively used in CLR source code?

Yes. For example, a quick search for `ULONG32` shows 4670 instances under src\coreclr.

File: src/coreclr/pal/inc/pal_mstypes.h (hunk 1 above)
Comment (jkotas) [reply src/coreclr/pal/inc/pal_mstypes.h]: They cannot. Changing them leads to build breaks, and fixing those build breaks would require introducing more complicated ifdefs elsewhere (or larger refactoring that I am not planning to do in this PR).

**Comment by jkotas** [comment]
This is ready for review

**Review by AaronRobinsonMSFT — APPROVED** [review]
:shipit:

**Pull request merged**